src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -1049,9 +1048,7 @@ bool primitive_inst::update_impl(bool use_async_compilation) {
                 if (!can_be_optimized())  {
                     if (!is_current_impl_dynamic)
                         _impl = std::move(_dynamic_impl);
-                    auto new_impl_params = _impl->canonicalize_shapes(*_impl_params);
-                    _impl->update_dispatch_data(new_impl_params);
-                    update_shape_info_tensor(new_impl_params);
+                    _impl->update(*this, *_impl_params);
```

Comment:
nit: how about renaming it to something like `update_params()` / `update_impl_configuration()` / `reconfigure_impl()` to be more descriptive?